src/data_validation/validation.py

```diff
@@ -0,0 +1,73 @@
+import os
+import toml
+
+from src.utils.hdfs_mods import read_hdfs_csv as read_data
+
+datafilepath = "/ons/rdbe_dev/Frozen_Group_Data2021_244_Headers.csv"
+
+
+def load_schema(filePath: str = "./config/DataSchema.toml") -> dict:
+    """Load the data schema from toml file into a dictionary
+
+    Keyword Arguments:
+        filePath -- Path to data schema toml file
+        (default: {"./config/DataSchema.toml"})
+
+    Returns:
+        A dict: dictionary containing parsed schema toml file
+    """
+    file_exists = os.path.exists(filePath)
+
+    # Check if DataSchema.toml exists
+    if not file_exists:
+        return file_exists
+    else:
+        # Load toml data schema into dictionary
+        toml_string = toml.load(filePath)
+    return toml_string
+
+
+def check_data_shape(
+    dataFile: str = datafilepath,
+    filePath: str = "./config/DataSchema.toml",
+    numCols: int = 93,
+) -> bool:
+    """Compares the shape of the data and compares it to the shape of the toml
+    file based off the data schema. Returns true if there is a match and false
+    otherwise.
+
+    Keyword Arguments:
+        dataFile -- Path to data file to compare (default: {datafilepath})
+        filePath -- Path to schema dictionary file
+        (default: {"./config/DataSchema.toml"})
+        numCols -- Number of columns in data (default: {93})
+
+    Returns:
+        A bool: boolean, True if number of columns is as expected, otherwise False
+    """
+
+    cols_match = False
+
+    # Read data file
+    data = read_data(dataFile)
+
+    # Convert it to dictionary
+    data_dict = data.to_dict()
+
+    # Load toml data schema into dictionary
+    toml_string = load_schema(filePath)
+
+    # Create a 'shared key' dictionary
+    shared_items = {k: toml_string[k] for k in toml_string if k in data_dict}
+
+    # Compare number of 'columns' in data to data schema
+    if len(shared_items) == len(toml_string):
+        cols_match = True
+    else:
+        cols_match = False
```

Comment:
This would work but is a very round-about way to do things. 

Would not something like
```python
if df.shape[0] == len(toml_dict):
     return True
else:
    return False
```

...work?

Reply:
Yes it does work and is cleaner than the shared_items comparison. I have amended this in the latest commit.